src/gpu/intel/compute/device_info.cpp

```diff
@@ -70,6 +70,31 @@ bool device_info_t::is_integrated() const {
     return ngen::getPlatformType(family) == ngen::PlatformType::Integrated;
 }
 
+std::string device_info_t::get_cl_ext_options() const {
+    using namespace compute;
+
+    std::string opts;
+    for (uint64_t i_ext = 1; i_ext < (uint64_t)device_ext_t::last;
+            i_ext <<= 1) {
+        auto ext = (device_ext_t)i_ext;
+
+        // Use real GPU extensions
+        if (!has(ext)) continue;
```

Comment:
Currently, the extensions info is initialized using SYCL aspects, which is not accurate because SYCL aspects != OpenCL extensions and was a temporary solution. Instead, we should use `ext_oneapi_supports_cl_extension` ([spec](https://github.com/intel/llvm/blob/f270fc72d2a4fb915a9c6bba0e7497a344969312/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler_opencl.asciidoc#querying-supported-features-and-extensions)) to query the actual OpenCL extensions inside `gpu::intel::sycl::device_info_t::init_extensions()`. 

Once implemented, the `gpu::intel::sycl::compat::init_extensions()` function will not longer be needed.